### PR TITLE
My reference topic

### PR DESCRIPTION
**Pull request by radhikasundararaman24**

### CODE REVIEW

**Comment by radhikasundararaman24** [comment]
Merge my reference topic after review.

File: docs/api/params-for-successful-requests.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+## URL: /tasks
+
+| Parameters| Type|Description|
+|---|---|---|
+|user_id | Integer   | Indicates the user-specified numeric value associated with a new task.|
+| title | Alphanumeric | Indicates a brief note about the task that you want to add.|
+| description | Varchar | Description of the task.   |
+| due_date  | Datetime   | Completion date of the task.   |
+| warning  | Varchar  | Indicates the user-specified note that can impact a task.|
```
Comment (rbwatson) [inline docs/api/params-for-successful-requests.md]: REST APIs typically use [JSON values](https://restfulapi.net/json-data-types/). These look like SQL data types, which, if required, would need to be expressed in a JSON data type, with whatever restrictions necessary to make them compatible. e.g. a DATETIME SQL type, would be stored in JSON as a string, formatted as date: YYYY-MM-... Date values in JSON are often expressed in [ISO 8601 format](https://www.iso.org/iso-8601-date-and-time-format.html).

File: docs/api/params-for-successful-requests.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Parameters for successful requests
+
+This section highlights the required and optional parameters for the To-Do service.
+
+To get a successful response, at a minimum, you must specify the properties listed under each task. 
+
+# Add a new task
```
Comment (rbwatson) [inline docs/api/params-for-successful-requests.md]: In a markdown file, there can only be one H1 title, which is the topic title.

**Review by rbwatson — CHANGES_REQUESTED** [review]
I'm not sure about the nature of this topic.
It seems to be a collection of several topics in one file.
Can you provide some more context about this and where it would fit in the documentation, please?

Thanks!

**Comment by radhikasundararaman24** [comment]
> I'm not sure about the nature of this topic. It seems to be a collection of several topics in one file. Can you provide some more context about this and where it would fit in the documentation, please?
> 
> Thanks!

Hi there, 
I was thinking of a dedicated topic to reference the required elements for each request. Let me know if this would fit...

Thanks,
Radhika

File: docs/api/params-for-successful-requests.md (hunk 2 above)
Comment (radhikasundararaman24) [reply docs/api/params-for-successful-requests.md]: Got it! Will update..

File: docs/api/params-for-successful-requests.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# Parameters for successful requests
+
+This section highlights the required and optional parameters for the To-Do service.
```
Comment (rbwatson) [inline docs/api/params-for-successful-requests.md]:
```suggestion
This section highlights the required and optional parameters for the operation s of the To-Do service.
```

File: docs/api/params-for-successful-requests.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+To get a successful response, at a minimum, you must specify the properties listed under each task.
```
Comment (rbwatson) [inline docs/api/params-for-successful-requests.md]:
```suggestion
To get a successful response from the following operations,you must specify at least the `required` properties listed under each task. 
```

**Review by rbwatson — CHANGES_REQUESTED** [review]
I see what you're trying to do and added some suggestions to help support that goal.
Feel free to add more clarification, if necessary.

Also add a link to this topic from someplace that makes sense in the existing topics.

Thanks!

**Comment by radhikasundararaman24** [comment]
> I see what you're trying to do and added some suggestions to help support that goal. Feel free to add more clarification, if necessary.
> 
> Also add a link to this topic from someplace that makes sense in the existing topics.
> 
> Thanks!

Can I create this topic reference in the API reference docs section on this > https://uwc2-apidoc.github.io/to-do-service-public/
 page?

**Comment by radhikasundararaman24** [comment]
Review requested for pr-71

File: docs/api/params-for-successful-requests.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+
+
+## Add a new task
+| Property| Type|Description|
+|---|---|---|
+|user_id | Number   |Indicates the user-specified numeric value associated with a new task.|
+| title | String |Indicates a brief note about the task that you want to add.|
+| description | String | Description of the task.   |
+| due_date  | String (format: YYYY-MM-DD)   | Completion date of the task.   |
```
Comment (rbwatson) [inline docs/api/params-for-successful-requests.md]:
```suggestion
| due_date  | String (format: YYYY-MM-DDTHH:MM:SS)   | Completion date of the task.   |
```
Don't forget the time!

File: docs/api/params-for-successful-requests.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+| warning  | String  | Indicates the user-specified note that can impact a task.|
```
Comment (rbwatson) [inline docs/api/params-for-successful-requests.md]:
```suggestion
| warning  | number| When to remind the user of the task, relative to the `due_date`. Negative numbers indicate before the `due_date`, and positive numbers, after. |
```
Check the `task` documentation to confirm this change.

File: docs/api/params-for-successful-requests.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+# Parameters for successful requests
+
+This section highlights the required and optional parameters for the operations of the To-Do service.
+
+To get a successful response from the following operations, you must specify at least the `required` properties listed under each task. 
+
+
+
+## Add a new task
+| Property| Type|Description|
```
Comment (rbwatson) [inline docs/api/params-for-successful-requests.md:19]:
```suggestion

**HTTP method**: POST
**URL**: `<server>/task`

| Property| Type|Description|
```
All of these need an example of the URL to call to provide context for the properties in the table. For example, the properties in this table go in the request body, while for the "find" operations, the search parameter goes in the URL.

Maybe add a column for where the property is found, e.g. `query parameter` or `request body` or something like that?

**Review by rbwatson — CHANGES_REQUESTED** [review]
This is coming together.

I made some suggestions in the file.
And, be sure to link this to an existing topic so it can be opened.

Thanks!

**Comment by rbwatson** [comment]
> Can I create this topic reference in the API reference docs section on this https://uwc2-apidoc.github.io/to-do-service-public/
> page?

How about in the https://uwc2-apidoc.github.io/to-do-service-public/#api-reference-docs section which is in the `docs/index.md` file

**Comment by radhikasundararaman24** [comment]
Open pull request for #71..

**Review by rbwatson — APPROVED** [review]
Rebased and squashed for merge

**Pull request merged**